index.html

```diff
@@ -957,6 +959,191 @@ <h4>Error Response</h4>
+              </td>
+            </tr>
+            <tr>
+              <td><code>timestamp</code></td>
+              <td>string</td>
+              <td>Optional</td>
+              <td>A timestamp in date-time format [[RFC3339]] set to the time the property readings took place.</td>
+            </tr>
+          </tbody>
+        </table>
+        <pre class="example" title="writeallproperties response message">
+          {
+            "thingID": "https://mythingserver.com/things/mylamp1",
+            "messageID": "74022347-72c2-4fa1-9e34-1a662bdff85e",
+            "messageType": "response",
+            "operation": "writeallproperties",
+            "values": {
+              "on": true,
+              "level": 75
+            },
+            "timestamp": "2025-08-13T17:11:00.531Z",
+            "correlationID": "551aaa52-ff95-4fb5-a657-2b5f430d47ff"
+          }
+        </pre>
+      </section>
+      <section id="writeallproperties-error-response">
+        <h4>Error Response</h4>
+        <p>If a <a>Thing</a> receives a <code>writeallproperties</code> request message from a <a>Consumer</a> which
+          does not contain values for all writeable properties, contains an invalid <a>Property</a> name, contains the name
+          of a <a href="https://www.w3.org/TR/wot-thing-description/#td-vocab-readOnly--DataSchema"><code>readOnly</code></a>
+          [[wot-thing-description11]] <a>Property</a>, or contains a value which does not conform to the data schema of the
+          corresponding PropertyAffordance then it MUST send an <a href="#error-response">error response</a>
+          to the <a>Consumer</a> with <code>status</code> set to <code>400</code>.
+        </p>
+        <pre class="example" title="Example writeallproperties Bad Request error response message">
+          <code>
+            {
+              "thingID": "https://mythingserver.com/things/mylamp1",
+              "messageID": "6dcfce6a-b0cd-48e7-992a-b96d5cb2b872",
+              "messageType": "response",
+              "operation": "writeallproperties",
+              "error": {
+                "status": "400",
+                "type": "https://w3c.github.io/web-thing-protocol/errors#400",
+                "title": "Bad Request"
+                "detail": "No property found with the name 'volume'"
+              }
+              "timestamp": "2025-08-13T18:31:43.531Z",
+              "correlationID": "551aaa52-ff95-4fb5-a657-2b5f430d47ff"
+            }
+          </code>
+        </pre>
+        <p>If a <a>Thing</a> receives a <code>writeallproperties</code> request message from a <a>Consumer</a> with
+          a <code>values</code> member containing valid names and values for all writeable <a>Properties</a>, but the
+          writing of at least one of the properties unexpectedly fails, then it MUST send an
+          <a href="#error-response">error response</a> to the <a>Consumer</a> with <code>status</code> set to
+          <code>500</code>.
```

Comment:
A status of 500 'internal server error' would suggest something is wrong on the Thing. However, the Thing is working exactly as intended. What is wrong is the consumer sending a request that doesn't meet the requirements. Would error [400 - Bad Request](https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Status/400) be more appropriate?

Reply:
Actually if you re-read the assertion, this error case is for when the Consumer sends a _valid_ request but something goes wrong on the Thing, so I think 500 is appropriate here.